sdk/core/azure-core/inc/azure/core/md5.hpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Utility functions to help compute the MD5 hash value for the input binary data.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+namespace Azure { namespace Core {
+
+  /**
+   * @brief Represents the class for the MD5 hash function which maps binary data of an arbitrary
+   * length to small binary data of a fixed length.
+   */
+  class Md5 {
+  public:
+    /**
+     * @brief Construct a default instance of @Md5.
+     */
+    Md5();
```

Comment:
```suggestion
    explicit Md5();
```

Reply:
When do we use `explicit` for ctors? Do all user-defined parameterless ctor need it, or do we want to add it for all ctors even the ones that accept parameters?

Reply:
We want to only use explicit ctors. @RickWinter  did you create a guideline for this?

Reply:
But explicit for constructors without any parameter is almost useless. You may want to make that an exception.

Reply:
well, it does not hurt at all, but yet keep a general pattern across all ctors. 
Consider it as a reminder for any future where you want to add one more ctor in here, you would quickly have the explicit over there :).

Reply:
yeah, but having a guideline suggesting some useless thing is not a good idea.